mmpose/evaluation/metrics/keypoint_partition_metric.py

```diff
@@ -0,0 +1,132 @@
+# Copyright (c) OpenMMLab. All rights reserved.
+
+from collections import OrderedDict
+from copy import deepcopy
+from typing import Optional, Sequence
+
+import numpy as np
+from mmengine.evaluator import BaseMetric
+
+from mmpose.registry import METRICS
+
+
+@METRICS.register_module()
+class KeypointPartitionMetric(BaseMetric):
+    """Wrapper metric for evaluating pose metric on user-defined body parts.
+
+    Sometimes one may be interested in the performance of a pose model on
+    certain body parts rather than on all the keypoints. For example,
+    ``CocoWholeBodyMetric`` evaluates coco metric on body, foot, face,
+    lefthand and righthand. However, ``CocoWholeBodyMetric`` is not flexible
+    enough to be applied to arbitrary custom datasets. Therefore, we provide
+    this wrapper metric.
+
+    Supported metrics:
+        ``CocoMetric``  all keypoint ground truth should be stored in
+            `keypoints` not other data fields.
+        ``PCKAccuracy`` the required data fields should be provided in
+            annotation files, such as head bbox, etc.
+        ``AUC``
+        ``EPE``
```

Comment:
The descriptions of supported metrics are inadequate; BTW, it would be beneficial to include a type check on the wrapped metric in the `__init__` function.

Reply:
I think AUC and EPE can be used without any condition, so I didn't write anything there. It is a good idea to do a type check.